src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -19,5 +27,116 @@ public Task Get()
 
             return Task.CompletedTask;
         }
+
+        [RavenAction("/info/tcp/pull", "GET", AuthorizationStatus.UnauthenticatedClients)]
+        public async Task GetPullReplicationInfo()
+        {
+            var databaseName = GetStringQueryString("databaseName");
+            var databaseGroupId = GetStringQueryString("databaseGroupId");
+            var definitionName = GetStringQueryString("definitionName");
+
+            var nodes = GetResponsibleNode(databaseName, databaseGroupId, definitionName);
+
+            // TODO: cache the tcp info for each node?
+            DynamicJsonArray output = new DynamicJsonArray();
+            var clusterTopology = ServerStore.GetClusterTopology();
+            var tasks = new List<Task<TcpConnectionInfo>>();
+            foreach (var node in nodes)
+            {
+                if (node == ServerStore.NodeTag)
+                {
+                    output.Add(Server.ServerStore.GetTcpInfoAndCertificates(HttpContext.Request.GetClientRequestedNodeUrl()));
+                }
+                else
+                {
+                    var nodeUrl = clusterTopology.GetUrlFromTag(node);
+                    tasks.Add(GetNodeTcpInfo(nodeUrl, databaseName));
+                }
+            }
+
+            await Task.WhenAll(tasks);
+            foreach (var task in tasks)
+            {
+                output.Add(task.Result.ToJson());
+            }
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                context.Write(writer, new DynamicJsonValue
+                {
+                    ["Results"] = output
+                });
+            }
+        }
+
+        private async Task<TcpConnectionInfo> GetNodeTcpInfo(string nodeUrl, string databaseName)
```

Comment:
Why are you already getting the `TcpConnectionInfo` here? Let the client actually do that.
The response from here should be just the node urls as `http/s`

Reply:
The reason this concerns me is that you are using the internal channels, and the remote server might be using a different network.

Reply:
The reason here is that `/info/tcp` require `ValidUser` which the edge don't have.

Reply:
So have another endpoint that would do the validation properly.
The problem of getting internal ip is something we want to avoid